File: LeetCode_Marathons/LCM5/Easy/Plus_One/a.go
func plusOne(digits []int) []int {
    n := len(digits)
    digits[n - 1]++
    for i := n - 1; i >= 0; i-- {
        if digits[i] >= 10 {
            digits[i] = 0
            if i != 0 {
                digits[i - 1]++
            } else {
                return append([]int{1}, digits...)
            }
        } else {
            return digits
        }
    }
    return []int{}
}

File: LeetCode_Marathons/LCM5/Easy/Plus_One/a.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        n: int = len(digits)
        digits[n - 1] += 1
        for i in range(n - 1, -1, -1):
            if digits[i] == 10:
                digits[i] = 0
                if i != 0: digits[i - 1] += 1
                else: return [1] + digits
            else:
                return digits